pkg/proxy/topology.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxy
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func FilterTopologyEndpoint(currentNodeName types.NodeName, nodeMap NodeMap, topologyKeys []string, endpoints []Endpoint) []Endpoint {
```

Comment:
Needs a good comment

Reply:
nit - FilterTopologyEndpoints - being plural matters to readabiltiy

Reply:
Comments was added